server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -963,18 +963,24 @@ public void commitSegmentInfos(SegmentInfos latestSegmentInfos, long maxSeqNo, l
         }
     }
 
+    public DirectoryFileTransferTracker getDirectoryFileTransferTracker() {
+        return directory.getDirectoryFileTransferTracker();
+    }
+
     /**
      * A store directory
      *
      * @opensearch.internal
      */
     static final class StoreDirectory extends FilterDirectory {
-
         private final Logger deletesLogger;
 
+        public final DirectoryFileTransferTracker directoryFileTransferTracker;
```

Comment:
This is now going to apply to every use of the local store.copyFrom. I'm not sure this belongs here? Can we can wrap and collect these required stats directly in the store directory or related components?

Reply:
I wonder if we should think about making this tracking optional by way of an expert setting?